samples/bluetooth/peripheral_with_multiple_identities/src/main.c

```diff
@@ -0,0 +1,179 @@
+
+	printk("disconnected %s, reason %u\n", addr_str, reason);
+
+	struct bt_conn_info connection_info;
+	int err;
+	err = bt_conn_get_info(conn, &connection_info);
+
+	if (err) {
+		printk("Failed to get conn info (err %d)\n", err);
+		return;
+	}
+
+	/* Get the ID of the disconnected advertiser. */
+	uint8_t id_current = connection_info.id;
+	printk("Advertiser %d disconnected\n", id_current);
+
+	/* Restart the advertiser by submitting its work to the workqueue. */
+	k_work_submit(&advertisers[id_current].work);
+}
+
+static struct bt_conn_cb conn_callbacks = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static void start_connectable_advertiser(struct k_work *work)
+{
+	int err;
+
+	struct advertiser_info *current_adv_info =
+		CONTAINER_OF(work, struct advertiser_info, work);
+
+	if (current_adv_info->adv) {
+		err = bt_le_ext_adv_start(current_adv_info->adv, BT_LE_EXT_ADV_START_DEFAULT);
+		if (err) {
+			printk("Failed to start advertising set (err %d)\n", err);
+			return;
+		}
+	} else {
+		printk("Advertiser not setup correctly\n");
+		return;
+	}
+
+	printk("Advertiser %d successfully started\n", current_adv_info->id);
+}
+
+static int setup_advertiser(uint8_t id_adv)
+{
+	/* Initialize the parameters for each connecable advertiser. */
+	struct bt_le_adv_param adv_param =
+		BT_LE_ADV_PARAM_INIT(BT_LE_ADV_OPT_CONNECTABLE,
+				     BT_GAP_ADV_SLOW_INT_MIN,
+				     BT_GAP_ADV_SLOW_INT_MAX,
+				     NULL);
+	
+	size_t id_count = 0xFF;
+	int err;
+	
+	/* Retrieve the number of currently configured identities. */
+	bt_id_get(NULL, &id_count);
+	if (id_adv == id_count) {
+		int id;
+		
+		/* Creates a new identity, with a new random static address and random IRK. */
+		id = bt_id_create(NULL, NULL);
+		if (id < 0) {
+			printk("Create id failed (%d)\n", id);
+			if (id_adv == 0) {
+				id_adv = CONFIG_BT_EXT_ADV_MAX_ADV_SET;
+			}
+			id_adv--;
+		} else {
+			printk("New id: %d\n", id);
+		}
+	}
+
+	printk("Using current id: %u\n", id_adv);
+	adv_param.id = id_adv;
+	advertisers[id_adv].id = id_adv;
+	
+	/* Create a new advertising set. */
+	err = bt_le_ext_adv_create(&adv_param, NULL, &advertisers[id_adv].adv);
+	if (err) {
+		printk("Failed to create advertiser set (err %d)\n", err);
+		return err;
+	}
+
+	/* Generate a new name to differentiate between advertisers. */
+	snprintf(device_name_buf, sizeof(device_name_buf), "%s %d", CONFIG_BT_DEVICE_NAME, id_adv);
+	printk("Created %s: %p\n", device_name_buf, &advertisers[id_adv].adv);
+
+	/* Set the advertising data. */
+	struct bt_data adv_data[] = {
+		BT_DATA(BT_DATA_NAME_COMPLETE, device_name_buf, strlen(device_name_buf)),
+	};
+
+	err = bt_le_ext_adv_set_data(advertisers[id_adv].adv, adv_data, ARRAY_SIZE(adv_data), NULL, 0);
+	if (err) {
+		printk("Failed to set advertising data (err %d)\n", err);
+		return err;
+	}
+
+	/* Clear the global buffer. */
+	memset(device_name_buf, 0, sizeof(device_name_buf));
```

Comment:
We shouldn't need to clear the advertising buffer
```suggestion

```

Reply:
Unnecessary if I remove the global buffer which I have now :)